### DISCUSSION TITLE
[Dart] Problem with OAuth (Google and Facebook)

### DISCUSSION THREAD

**Discussion by Belvin-04** [pocketbase/pocketbase#5874] [Q&A] [answered]
I am following the documentation for google and facebook auth.
Facebook:
`
final authData = await pb.collection('users').authWithOAuth2('facebook', (url) async {
      await launchUrl(url);
});
`
Google:
`
final authData = await pb.collection('users').authWithOAuth2('google', (url) async {
      await launchUrl(url);
});
`
I get the following in the logs:
INFO GET /api/collections/users/auth-methods
DEBUG Realtime connection established.
└─ map[clientId:KXE86fM18k5GI7hFhrGDVZjCqXRYTbFNnvv4vHa9]
INFO GET /api/collections/users/auth-methods
DEBUG Missing OAuth2 state parameter

Am i missing something during the implementation or is there any other problem?

**Comment by ganigeorgiev** [comment, marked as answer]
It works correctly for me when testing with Dart cli and PocketBase v0.22.25/Dart SDK v0.18.1.

If you are testing with Flutter Web note that the "all-in-one" `authWithOAuth2()` call require establishing a realtime connection and streamed responses are not supported on the web at the moment out of the box with the standard `dart-lang/http` client. You can find a workaround in https://github.com/pocketbase/dart-sdk?tab=readme-ov-file#limitations.

If that's not the case, please provide a more complete code sample and information about your environment and how you are testing.

**Reply by ganigeorgiev** [reply]
To rule out Google/Facebook OAuth2 app misconfiguration you can try first testing with the following minimal Dart example:

```dart
// example.dart
import 'dart:io';

import 'package:pocketbase/pocketbase.dart';

void main() async {
  final pb = PocketBase('http://127.0.0.1:8090');

  final authData = await pb.collection('users').authWithOAuth2('google', (url) async {
    // calls a Linux shell command because `launchUrl` requires Flutter dependency
    // 
    // if you are testing on mac replace `xdg-open` with `open`
    Process.run('xdg-open', [url.toString()]);
  });

  print(authData);
}
```

Run the above with `dart run example.dart`.

**Reply by Belvin-04** [reply]
Yes, I was testing with Flutter Web. The workaround you provided in the first reply worked perfectly.
Thank you very much...! :+1:
